container/build_tar.py

```diff
@@ -191,18 +231,28 @@ def add_deb(self, deb):
     Raises:
       DebError: if the format of the deb archive is incorrect.
     """
+    pkg_data_found = False
+    pkg_metadata_found = False
     with archive.SimpleArFile(deb) as arfile:
       current = arfile.next()
-      while current and not current.filename.startswith('data.'):
+      while current:
+        name, ext = os.path.splitext(current.filename)
+        if name == 'data':
+          pkg_data_found = True
+          # Add pkg_data to image tar
+          with self.write_temp_file(suffix=ext, data=current.data) as tmpfile:
+            self.add_tar(tmpfile)
+        elif name == 'control':
+          pkg_metadata_found = True
+          # Add metadata file to image tar
+          with self.write_temp_file(suffix=ext, data=current.data) as tmpfile:
+            self.add_pkg_metadata(metadata_tar=tmpfile, deb=deb)
         current = arfile.next()
-      if not current:
-        raise self.DebError(deb + ' does not contains a data file!')
-      tmpfile = tempfile.mkstemp(suffix=os.path.splitext(current.filename)[-1])
-      with open(tmpfile[1], 'wb') as f:
-        f.write(current.data)
-      self.add_tar(tmpfile[1])
-      os.remove(tmpfile[1])
 
+    if not pkg_data_found:
+      raise self.DebError(deb + ' does not contains a data file!')
+    if not pkg_metadata_found:
```

Comment:
I assume packages must have metadata?

Reply:
I would just like to avoid having this be a breaking change.

Reply:
I read up on how packages get uploaded to dpkg and they have to have control file.
https://www.debian.org/doc/manuals/maint-guide/dreq.en.html#control